ddtrace/tracer/tracer_test.go

```diff
@@ -1095,6 +1098,24 @@ func decode(p *payload) (spanLists, error) {
 	return traces, err
 }
 
+func (t *dummyTransport) waitFlush(n int, timeout time.Duration) {
+	a := time.After(timeout)
+	for {
+		select {
+		case <-a:
+			panic("timed out waiting for flush")
```

Comment:
The most correct way would be to do a `flushAndWait(t *testing.T, n int)` and then use `t.Fatal` here. I'm afraid `panic` could be misleading and indicate a programming error, which is usually the case for its use. Do you think that'd be ok to do so?

Reply:
I'll add a `*testing.T` argument so we can call `Fatal`. That will be a better failure mode for tests.